src/test/java/com/puppycrawl/tools/checkstyle/DetailAstImplTest.java

```diff
@@ -134,6 +135,22 @@ public void testGetChildCount() throws Exception {
         assertEquals(firstLevelA, firstLevelB.getPreviousSibling(), "Invalid previous sibling");
     }
 
+    @Test
+    public void testHasChildren() throws Exception {
```

Comment:
Do we need this test ?

We try to avoid pure UTs as much as possible.

Reply:
done

Reply:
now `./.ci/pitest.sh pitest-common-2` fails

Reply:
We need this test. I think we should keep all test for API. For an utility class a pure UT test is redundant, but this is the part of our API.

Reply:
As this test is for technical coverage, we need to add comment over it to explain reason.

Reply:
The whole class `DetailAstImplTest` is for technical coverage. As many other tests. It's good to add this description to all of them, but it should be a separate issue.

Reply:
@romani We had a discussion about this when we created DetailASTImpl (I believe). Some random Check covered a line in the AST. It was some completely random test and it didn't make sense how it was proof the AST API method was shown to be working correctly. Thats when we said we would force API to have their own UTs.

Reply:
this happens because of very limited test execution scope .... we did this for performance reason.
https://github.com/checkstyle/checkstyle/blob/master/pom.xml#L2722

from pitest report:
`return getFirstChild() != null;`
```
1. replaced boolean return with true for com/puppycrawl/tools/checkstyle
/DetailAstImpl::hasChildren → NO_COVERAGE
2. negated conditional → NO_COVERAGE
3. removed conditional - replaced equality check with false → NO_COVERAGE
4. removed conditional - replaced equality check with true → NO_COVERAGE
5. replaced return of integer sized value with (x == 0 ? 1 : 0) → NO_COVERAGE
```

Reply:
This test method is using reflection ... it must have comment to explain it (we agreed on this a long time ago). Please also add some sentence that it is required only because to satisfy certain pitest profile .

one day it will change so we will have details on why it is required.

Reply:
This test should not use any reflection. Fixed.